### DISCUSSION TITLE
Designing elements in XC

### DISCUSSION THREAD

**Discussion by ebrahimraeyat** [xcfem/xc#73] [Q&A]
Hi @lcpt. I studied some verification test and xc examples, but I want to know is it possible to design elements with XC, I mean designing concrete beam for example, or check some codes provision like joint shear in concrete structure? Thanks.

![joint_shear](https://user-images.githubusercontent.com/8196112/103170755-0e5b7d00-485c-11eb-9259-9e9874cb405d.png)

**Comment by lcpt** [comment]
Hi @ebrahimraeyat 

Do you mean write a new "specialized" kind of element that makes easier to model a particular part of the structure?

That will be the case of the Joint2D and Joint3D elements inherited from OpenSees for example (you can't use them in XC yet but the process of exposing them to Python and write some test is quite straightforward).

**Reply by ebrahimraeyat** [reply]
No, I mean the design process, not element for model. For example designing concrete beams to obtain longitudinal  and shear rebars for beams and so on.
I don't mean Joint2D or Joint3D for checking joint shear. I will explain more this one.

**Reply by lcpt** [reply]
We usually design the structural members by means of simple calculations, and then we use XC to check all the limit states under all the load combinations. For an experienced user it takes at most two or three iterations to achieve a good design.

At present there is no "automatic design" routines that you can use to do this.

**Comment by ebrahimraeyat** [comment]
Thanks, @lcpt.
Can you please name some limit state in concrete structures? For example for beams.
Let say for beams. After analysis, we must design the beams, columns most of the time only must be checked for demand / capacity ratio that seems applied in XC interaction diagram. But for beams we need to design all beams in all over the length of beams (usually a defined station for example each 10 centimeters) for all load combinations. Then we can obtain the area of desired rebar in all over the beam. But now how do you design your beams?
Can you prepare some series video about designing in XC? I mean quick overview, not detailing. Thanks.

**Comment by lcpt** [comment]
Hi @ebrahimraeyat 

Well, XC is more like a toolbox that allows the user to organize things at his convenience depending on the problem. There are always multiple ways you can follow to solve a problem.

Nevertheless, for the beam design you mentioned the normal path is as follows:

- Predimensioning of the reinforcement along the beam. As a result, you have a set of RC sections for the different parts of the beam (mid span, ...).
- Prepare a linear elastic finite element model of the beam with a suitable element length (let's say 0.25 m) and assign a reinforced concrete section to each element in the beam.
- Run the analysis and the program compute the worst capacity factors in bending and shear.
- Examine the results and decide if the reinforcement needs to be modified or not.

I can prepare an example showing how to do this (sorry I've not the time to prepare a video, moreover I think a video is more useful when you want to show how to use the program menus and dialogs; here it's all about scripting).

**Reply by ebrahimraeyat** [reply]
> Hi @ebrahimraeyat
> I can prepare an example showing how to do this (sorry I've not the time to prepare a video, moreover I think a video is more useful when you want to show how to use the program menus and dialogs; here it's all about scripting).

Thanks, @lcpt. But this way is very time-consuming for a real world project, isn't it? I mean analysis structure without reinforcement for beam and after analysis with obtaining Mu and Vu we can design reinforcement. What is your opinion?

**Comment by lcpt** [comment]
I don't make an analysis of the structure without reinforcement, there is no need to do so. Instead, I make some extremely simple calculations (no finite element model at all or, maybe, some very simple 2D models) to pre-dimensioning the slabs, the beams, the columns... and its reinforcement. That helps me to get a deeper understanding of my design (and to modify it if needed) and then, only when I am satisfied with it, I start creating the finite element model for the whole structure.

Nevertheless, it's always possible to write the routines to emulate the "automatic design" approach.

**Comment by ebrahimraeyat** [comment]
Thanks, @lcpt .I will examine it soon.
